compiler/pragmas.nim

```diff
@@ -655,13 +655,43 @@ proc pragmaGuard(c: PContext; it: PNode; kind: TSymKind): PSym =
   else:
     result = qualifiedLookUp(c, n, {checkUndeclared})
 
+proc semCustomPragma(c: PContext, n: PNode): PNode =
+  assert(n.kind in {nkIdent, nkExprColonExpr})
+  result = newNodeI(nkCall, n.info)
+  if n.kind == nkIdent:
+    result.add n 
+  elif n.kind == nkExprColonExpr and n[1].kind == nkBracket:
+    # pragma: [arg1, arg2] -> pragma(arg1, arg2)
+    result.add n[0]
+    result.sons &= n[1].sons
```

Comment:
I don't like direct 'sons' accesses but have no better solution right now.

Reply:
Now I have: Write a helper proc in ``ast.nim`` that does the required transformations.